### PR TITLE
Add `RepeatedAnnot` to `Reflection.defn`

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/reflect/CompilerInterface.scala [hunk 1]
```diff
@@ -1478,8 +1478,14 @@ trait CompilerInterface {
   def Definitions_Array_length: Symbol
   def Definitions_Array_update: Symbol
 
+  /** A dummy class symbol that is used to indicate repeated parameters
+   *  compiled by the Scala compiler.
```
Comment (AleksanderBG) [inline library/src/scala/tasty/reflect/CompilerInterface.scala:1482]: Can we clarify this or make an issue to clarify this in the future? Does this comment want to convey that this class is only used for Scalac repeated parameters?

Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/CompilerInterface.scala:1482]: I copied the original comment to the interface. We can revise this documentation later.

Comment (AleksanderBG) [reply library/src/scala/tasty/reflect/CompilerInterface.scala:1482]: So can we make an issue for it? Or are you keeping track of such things in a different way?

Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/CompilerInterface.scala:1482]: We will need to do a pass over everything in the API and probably rewrite/improve most of the documentation. Singling a few out does not seem to be worth it.

**Pull request merged**